sql/handler.cc

```diff
@@ -2658,9 +2660,13 @@ static my_bool xarecover_handlerton(THD *unused, plugin_ref plugin,
         */
         if (info->mem_root)
         {
-          // remember "full" xid too when it's not in mysql format
+          // remember "full" xid too when it's not in mysql format.
+          // Also record the server_id of the transaction. It will be used for
```

Comment:
I think a substitution like
  s/the server_id of the transaction /the transaction's original server_id if it's different from the current server_id/
reads better. Feel free to polish it, just two server_id:s should be put in contrast here.